lib/adapters/mongodb/index.js

```diff
@@ -55,7 +55,7 @@ module.exports = function (options) {
         return model.findById(req.params.id).lean().exec()
             .then((resources) => {
                 if (!resources) {
-                    return Boom.notFound()
+                    throw Boom.notFound()
```

Comment:
To trigger promise reject instead of resolve.
1 gru 2015 1:42 PM "Mehdi Avdi" <notifications@github.com> napisał(a):

> In lib/adapters/mongodb/index.js
> <https://github.com/agco/hapi-harvester/pull/11#discussion_r46272892>:
>
> > @@ -55,7 +55,7 @@ module.exports = function (options) {
> >          return model.findById(req.params.id).lean().exec()
> >              .then((resources) => {
> >                  if (!resources) {
> > -                    return Boom.notFound()
> > +                    throw Boom.notFound()
>
> why are we throwing here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/agco/hapi-harvester/pull/11/files#r46272892>.
>